src/app/content/highlights/hooks/sharedHooks.ts

```diff
@@ -0,0 +1,17 @@
+import { push } from '../../../navigation/actions';
+import * as navigation from '../../../navigation/selectors';
+import { getQueryForParam } from '../../../navigation/utils';
+import { AppServices, MiddlewareAPI } from '../../../types';
+import { modalQueryParameterName } from '../../constants';
+import { getContentParams } from '../../utils/urlUtils';
+
+export const openModal = (modalUrlName: string) => (services: MiddlewareAPI & AppServices) => () => {
```

Comment:
i don't think this has to be in `highlights` i think we can put it in `navigation`.

i see that you're using some pretty specific logic in your dispatch, but i think that could be replaced with generic logic if we normalize all the route params in the navigation state you could simply use those and replace the modal parameter

Reply:
I've moved it to the `navigation/hooks` and created separate files for `openModal` and `closeModal` hooks

> i see that you're using some pretty specific logic in your dispatch, but i think that could be replaced with generic logic if we normalize all the route params in the navigation state you could simply use those and replace the modal parameter

Do you mean to add an action like `pushParam('modal', 'MH')` that would be transformed into proper history action or do you mean to change how we deal with url params entirly so we can dispatch an action that changes the single param and associated hook would update the url?

Reply:
i mean that when navigation events are pushed they have parameters that are embedded/extracted from the url, those parameters aren't currently stored in the navigation state, they are just available for whoever needs them to pick them up.

currently you are deriving your new prams by re-computing your content params from the state, but what you really want is  "whatever my current params are but without the modal part of the query"

we could accomplish this in a generic way by making the navigation reducer add route params to navigation state as it does with other location parameters like query. then, instead of re-computing your parameters from the content state, which is content specific logic, you could use generic logic, of just taking what is in your navigation state and removing the modal part of the query